packages/editor/src/components/post-title/post-title-rich.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useState } from '@wordpress/element';
+import { ENTER } from '@wordpress/keycodes';
+import { useDispatch } from '@wordpress/data';
+import { pasteHandler } from '@wordpress/blocks';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import {
+	__unstableUseRichText as useRichText,
+	create,
+	toHTMLString,
+	insert,
+} from '@wordpress/rich-text';
+import { useMergeRefs } from '@wordpress/compose';
+import { __unstableStripHTML as stripHTML } from '@wordpress/dom';
+
+/**
+ * Internal dependencies
+ */
+import PostTypeSupportCheck from '../post-type-support-check';
+
+const PostTitleRich = forwardRef(
+	(
+		{ title, placeholder, onChange, onUpdate, className, setIsSelected },
+		ref
+	) => {
+		const [ selection, setSelection ] = useState( {} );
+
+		const { clearSelectedBlock, insertBlocks, insertDefaultBlock } =
+			useDispatch( blockEditorStore );
+
+		function onInsertBlockAfter( blocks ) {
+			insertBlocks( blocks, 0 );
+		}
+
+		function onSelect() {
+			setIsSelected( true );
+			clearSelectedBlock();
+		}
+
+		function onUnselect() {
+			setIsSelected( false );
+			setSelection( {} );
+		}
+
+		function onEnterPress() {
+			insertDefaultBlock( undefined, undefined, 0 );
+		}
+
+		function onKeyDown( event ) {
+			if ( event.keyCode === ENTER ) {
+				event.preventDefault();
+				onEnterPress();
+			}
+		}
+
+		function onPaste( event ) {
+			const clipboardData = event.clipboardData;
+
+			let plainText = '';
+			let html = '';
+
+			// IE11 only supports `Text` as an argument for `getData` and will
+			// otherwise throw an invalid argument error, so we try the standard
+			// arguments first, then fallback to `Text` if they fail.
+			try {
+				plainText = clipboardData.getData( 'text/plain' );
+				html = clipboardData.getData( 'text/html' );
+			} catch ( error1 ) {
+				try {
+					html = clipboardData.getData( 'Text' );
+				} catch ( error2 ) {
+					// Some browsers like UC Browser paste plain text by default and
+					// don't support clipboardData at all, so allow default
+					// behaviour.
+					return;
+				}
+			}
+
+			// Allows us to ask for this information when we get a report.
+			window.console.log( 'Received HTML:\n\n', html );
+			window.console.log( 'Received plain text:\n\n', plainText );
+
+			const content = pasteHandler( {
+				HTML: html,
+				plainText,
+			} );
+
+			event.preventDefault();
+
+			if ( ! content.length ) {
+				return;
+			}
+
+			if ( typeof content !== 'string' ) {
+				const [ firstBlock ] = content;
+
+				if (
+					! title &&
+					( firstBlock.name === 'core/heading' ||
+						firstBlock.name === 'core/paragraph' )
+				) {
+					onUpdate( stripHTML( firstBlock.attributes.content ) );
```

Comment:
Shouldn't this be reverted? (Introduced in #35825)
HTML is now allowed, so why strip?

Reply:
It's counterintuitive but it's intended. Why? The idea is that if you are pasting into the title you are very unlikely to want HTML to go into it.

A common action is copy/paste and accidentally including HTML in the pasted content. Then it ends up in the title.

With this PR at least you can _see_ that HTML, but this way it avoids it being added entirely.

I'm not strongly wedded to this change so if you feel it would be better to simply allow HTML and have the user remove it manually then I'd be happy to remove it.

Reply:
Improved code comments with explanation as to why this remains in place.